getting started and faq (#6490)
aws-amplify/docs: src/pages/gen2/start/quickstart/index.mdx

Let's make a quick change to our data model and add a `done` field which will be a `boolean` which can be set to `true` or `false` depending if our Todo list item is complete. Let's also add a priority field, which will be an enum. This field type will allow for just a few options to be stored - low, medium, or high.

> We've removed the to shorten the code below; however, you can chose to keep them for assistance building your app's connection to frontend resources.

Reviewer: ```suggestion
We've removed the default comments to shorten the code below; however, you can chose to keep them for assistance building your app's connection to frontend resources.
```

  },
});

// ...
```

Once you save your changes to the data model, your changes will be deployed in seconds within your cloud sandbox.

> Our Todo data model also has authorization set up. Our model has the `authorization` method chained to it, which you can add rules to. In our example, we are allowing an owner, or the person who creates the to do instance, to perform all actions on the data they own. We are also allowing all page viewers, including unauthenticated users,to read data. These can be modified, for example we could remove the `.to(['read'])` and allow all visitors to perform all actions on data. We could also add permissions for signed in users or users who belong to user groups such as `Admin`. You can learn more about all options for authorization in the [customize your auth rules](/build-a-backend/data/customize-authz/) section of the docs.

Reviewer 2: ```suggestion
Our Todo data model also has authorization set up. Our model has the `authorization` method chained to it, which you can add rules to. In our example, we are allowing an owner, or the person who creates the Todo instance, to perform all actions on the data they own. We are also allowing all page viewers, including unauthenticated users, to read data. These can be modified, for example we could remove the `.to(['read'])` and allow all visitors to perform all actions on data. We could also add permissions for signed in users or users who belong to user groups such as `Admin`. You can learn more about all options for authorization in the [customize your auth rules](/build-a-backend/data/customize-authz/) section of the docs.
```

Let's remove public access.

```js
.authorization([a.allow.owner()]),
```

> Then, you'll see the `defineData` function, which has our schema and authorization configuration passed in as arguments. We have an `API_KEY` set up in order to enable public access. Let's change our `defaultAuthorizationMode` to `userPool` instead so that the default is to use user authentication.

Reviewer: ```suggestion
Then, you'll see the `defineData` function, which has our schema and authorization configuration passed in as arguments. We have an `apiKey` set up in order to enable public access. Let's change our `defaultAuthorizationMode` to `userPool` instead so that the default is to use user authentication.
```

  return (
    <main>
      <h1>Hello, Amplify 👋</h1>
      <button onClick={async () => {
        // create a new Todo with the following attributes
        const { errors, data: newTodo } = await client.models.Todo.create({
          // prompt the user to enter the title
          content: window.prompt("title"),
          done: false,
          priority: 'medium'
        })
        console.log(errors, newTodo);
      }}>Create </button>

      <ul>
        {todos.map((todo) => (
          <li key={todo.id}>{todo.content}</li>
        ))}
      </ul>
    </main>
  );
}
```

> Create a couple to dos, then refresh the page to see them. You can also subscribe to new todos in your `useEffect` to have them live reload on the page.

Reviewer: ```suggestion
Create a couple todos, then refresh the page to see them. You can also subscribe to new todos in your `useEffect` to have them live reload on the page.
```

```js
useEffect(() => {
  const sub = client.models.Todo.observeQuery()
    .subscribe({
      next: ({ items, isSynced }) => {

>         setTodos(items)

Reviewer: ```suggestion
        setTodos([...items])
```

I'm almost sure you have to do something like this because otherwise React's reference won't be refreshed for the state.
